dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/BasicDeploymentForm.tsx

```diff
@@ -42,8 +42,31 @@ class BasicDeploymentForm extends React.Component<IBasicDeploymentFormProps> {
         // If hidden is a string, it points to the value that should be true
         return getValue(this.props.appValues, hidden) === true;
       case "object":
-        // If hidden is an object, inspect the value it points to
-        return getValue(this.props.appValues, hidden.value) === hidden.condition;
+        // Two type of supported objects
+        // A single condition: {value: string, path: any}
+        // An array of conditions: {conditions: Array<{value: string, path: any}, operator: string}
+        if (hidden.conditions?.length > 0) {
+          // If hidden is an object, a different logic should be applied
+          // based on the operator
+          switch (hidden.operator) {
+            case "and":
+              // Every value matches the referenced
+              // value (via jsonpath) in all the conditions
+              return hidden.conditions.every(
+                c => getValue(this.props.appValues, c.value) === c.path,
+              );
+            case "or":
+              // It is enough if the value matches the referenced
+              // value (via jsonpath) in any of the conditions
+              return hidden.conditions.some(
+                c => getValue(this.props.appValues, c.value) === c.path,
+              );
+            default:
+              return false;
```

Comment:
I would choose a default (probably `and` is better as a default) instead of returning always false if the operator is not defined.

Reply:
okey!